Packaging failures with Pytest 5.0

## Description

Pytest 5.0 was just released. It builds-in pytest-faulthandler and causes errors during packaging because we have pytest-faulthandler in requirements.txt.

Failing build: https://dev.azure.com/scikit-image/scikit-image/_build/results?buildId=761&view=logs&jobId=38fa3105-a3d6-58c5-2eec-10e695f0c1d9&taskId=a0d33157-d94d-587b-eae8-ff8bf4466631&lineStart=5880&lineEnd=5881&colStart=1&colEnd=1

We should either pin to 4.x or to 5.0+. I'd rather avoid complicated dependency logic doing different things for different pytest versions.

```

self = <skimage.util.tests.test_arraypad.ValueError3 testMethod=test_mode_not_set>

    def test_mode_not_set(self):
        arr = np.arange(30).reshape(5, 6)
        with testing.raises(TypeError):
>           pad(arr, 4)
E           Failed: DID NOT RAISE <class 'TypeError'>
```

This just seems like numpy decided to default padding to 0s if the mode wasn't set. I actually think that @stefanv  pushed for that change ^_^

--- a/requirements/test.txt
+++ b/requirements/test.txt
@@ -5,4 +5,3 @@ pytest-cov
 pytest-localserver
 flake8
 codecov
-pytest-faulthandler

